developer/src/kmdecomp/kmdecomp.cpp

```diff
@@ -152,9 +151,7 @@ BOOL LoadKeyboard(LPSTR fileName, LPKEYBOARD *lpKeyboard, LPBYTE *lpBitmap, DWOR
 	   ckbp->dwFileVersion > VERSION_MAX)
 	{
 		/* Old or new version -- identify the desired program version */
-    if((ckbp->dwFileVersion >= VERSION_160 && ckbp->dwCheckSum == 0) ||
-      /* #7222: We support a zero checksum in Keyman 16.0 and later */
-      VerifyChecksum(buf, &kbp->dwCheckSum, sz))
+    if(ckbp->dwFileVersion >= VERSION_160)
```

Comment:
In the same way as with kmanalyze, this change is wrong -- the `if() {` and corresponding `}` should be deleted entirely.

Reply:
fixed